### ISSUE TITLE
Modify Users Data Module to include payment methods

### ISSUE THREAD

**Issue by MmyTohsaka** [KanakaXVII/CS-546-WS-2023-Group-53#15]
**Requirements**

- Add user attribute for payment methods
- Add new function to add payment methods
- Add new function to modify payment methods
- Add new function to remove payment methods
- Add new function to view payment methods

**Comment by MmyTohsaka** [comment]
Yesterday, I added and tested the following functions:
- createPaymentMethod
- getPaymentMethodsByID
- deletePaymentMethodByID

I also made the following name changes for existing functions:
- get -> getUserByID
- create -> createUser

I made these changes to have more consistent and detailed names throughout the data files. I will work on the following function today. This should be the last function needed for payment methods in the `data/users.js` file. 
- updatePaymentMethod

**Comment by MmyTohsaka** [comment]
I created the `updatePaymentMethod` function to update the names of existing payment methods. I also modified the payment method objects to have an ObjectID and changed the DELETE operation to reference that Object ID instead of the name for queries.

I had to modify a bunch of throws in the helpers file because there were a lot that were broken (they were throwing strings only instead of strings plus a status code). 

Creating PR.
